src/components/SidebarPanel.js

```diff
@@ -0,0 +1,326 @@
+"use client"
+
+import { useEffect, useMemo, useState } from 'react';
+import { Menu, FileText, Volume2, Video, ChevronRight, X, ArrowLeft } from 'lucide-react';
+import { loadMediaIndex } from '@/lib/media';
+
+export default function SidebarPanel({ chapterTitle, locale, report, chapter }) {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [activeMediaPanel, setActiveMediaPanel] = useState(null); // 'audio' | 'videos' | null
+  const [selectedTrack, setSelectedTrack] = useState(null);
+
+  const [loading, setLoading] = useState(false);
+  const [media, setMedia] = useState({ audio: [], videos: [] });
+
+  // Load media for the current chapter
+  useEffect(() => {
+    let mounted = true;
+    (async () => {
+      setLoading(true);
+      try {
+        const data = await loadMediaIndex({ locale, report, chapter });
+        if (!mounted) return;
+        setMedia(data);
+      } finally {
+        if (mounted) setLoading(false);
+      }
+    })();
+    return () => {
+      mounted = false;
+    };
+  }, [locale, report, chapter]);
+
+  const menuItems = useMemo(() => ([
+    {
+      id: 'notes',
+      label: 'Notes',
+      icon: FileText,
+      color: 'text-gray-700 hover:bg-blue-50',
+      items: ['Chapter Notes', 'My Notes', 'Bookmarks']
+    },
+    {
+      id: 'audio',
+      label: 'Audio',
+      icon: Volume2,
+      color: 'text-gray-700 hover:bg-green-50',
+      items: media.audio
+    },
+    {
+      id: 'videos',
+      label: 'Videos',
+      icon: Video,
+      color: 'text-gray-700 hover:bg-red-50',
+      items: media.videos
+    }
```

Comment:
```suggestion
    {
      id: 'notes',
      label: t('panel.notes'),
      icon: FileText,
      color: 'text-gray-700 hover:bg-blue-50',
      items: ['Chapter Notes', 'My Notes', 'Bookmarks'],
      onClick: () => dispatch({ type: SET_ACTIVE_NOTES_PANEL_ACTION }),
    },
    {
      id: 'audio',
      label: t('panel.audios'),
      icon: Volume2,
      color: 'text-gray-700 hover:bg-green-50',
      onClick: () => dispatch({ type: SET_ACTIVE_AUDIOS_PANEL_ACTION }),
      items: state.media.audios
    },
    {
      id: 'videos',
      label: t('panel.videos'),
      icon: Video,
      color: 'text-gray-700 hover:bg-red-50',
      onClick: () => dispatch({ type: SET_ACTIVE_VIDEOS_PANEL_ACTION }),
      items: state.media.videos
    }
```
In JavaScript, `function` can be treated as a first-class citizen, a.k.a a variable or a property. You don't need the `handleMenuItemClick` function. Plus, you can avoid `useMemo` since passing from the server component to this component.